### PR TITLE
New/GUM 34 avatar capture

### PR DESCRIPTION
**Pull request by R-Mielamud**

### CODE REVIEW

**Comment by R-Mielamud** [comment]
Looks like this
Знімок екрана 2022-02-22 о 19 13 23

File: src/components/avatar-picker/avatar-picker.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { View } from 'react-native';
+import { Asset } from 'react-native-image-picker';
+import { ButtonMode, HeadingLevel } from 'src/common/enums';
```
Comment (katyakostikova) [inline src/components/avatar-picker/avatar-picker.tsx]:
```suggestion
import { Asset } from 'react-native-image-picker';

import { ButtonMode, HeadingLevel } from 'src/common/enums';
```

File: src/components/form/form-avatar-picker/form-avatar-picker.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { useFormikContext } from 'formik';
+import React from 'react';
+import { Asset } from 'react-native-image-picker';
+import { AvatarPicker } from 'src/components';
```
Comment (katyakostikova) [inline src/components/form/form-avatar-picker/form-avatar-picker.tsx]:
```suggestion
import { Asset } from 'react-native-image-picker';

import { AvatarPicker } from 'src/components';
```

**Review by katyakostikova — APPROVED** [review]

File: android/settings.gradle [hunk 3]
```diff
@@ -1,4 +1,6 @@
 rootProject.name = 'growupmobile'
+include ':react-native-image-picker'
+project(':react-native-image-picker').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-image-picker/android')
```
Comment (AleksandrTermenzhy) [inline android/settings.gradle:3]: Doesn't it autolink itself?

File: ios/Podfile [hunk 4]
```diff
@@ -12,6 +12,8 @@ target 'growupmobile' do
     :hermes_enabled => false
   )
 
+  pod 'react-native-image-picker', :path => '../node_modules/react-native-image-picker'
```
Comment (AleksandrTermenzhy) [inline ios/Podfile:15]: Same concern here

File: src/components/avatar-picker/avatar-picker.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        <MainButton
+          mode={ButtonMode.CONTAINED}
+          compact={true}
+          onPress={capture}
+        >
+          Take a photo
+        </MainButton>
+        <MainButton mode={ButtonMode.OUTLINED} compact={true} onPress={pick}>
+          Open gallery
+        </MainButton>
```
Comment (AleksandrTermenzhy) [inline src/components/avatar-picker/avatar-picker.tsx]: Let's use [menu](https://callstack.github.io/react-native-paper/menu.html) instead

File: src/helpers/image-capture/capture.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { PermissionsAndroid } from 'react-native';
+import { Asset, CameraType, launchCamera } from 'react-native-image-picker';
+
+const requestCameraPermission = async (): Promise<boolean> => {
```
Comment (AleksandrTermenzhy) [inline src/helpers/image-capture/capture.ts]:
```suggestion
const requestCameraPermissionAndroid = async (): Promise<boolean> => {
```

File: src/helpers/image-capture/capture.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    return permission === PermissionsAndroid.RESULTS.GRANTED;
+  } catch {
+    return false;
+  }
+};
+
+const captureSingleImage = async (
+  type?: CameraType
+): Promise<Asset | undefined> => {
+  if (!(await requestCameraPermission())) {
```
Comment (AleksandrTermenzhy) [inline src/helpers/image-capture/capture.ts]: Please extract this to a variable

File: src/helpers/image-capture/capture.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    );
+
+    return permission === PermissionsAndroid.RESULTS.GRANTED;
+  } catch {
+    return false;
+  }
+};
+
+const captureSingleImage = async (
+  type?: CameraType
```
Comment (AleksandrTermenzhy) [inline src/helpers/image-capture/capture.ts]: Don't we need default value here?

File: src/screens/onboarding/components/step-content/common.ts [hunk 9]
```diff
@@ -1,7 +1,10 @@
+import { Asset } from 'react-native-image-picker';
+
 const defaultAddUserInfoPayload = {
   firstName: '',
   lastName: '',
   position: '',
+  avatar: undefined as Asset | undefined,
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/components/step-content/common.ts:7]: Looks overcomplicated to me. Can we simplify type here?

File: src/components/avatar-picker/avatar-picker.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+import { captureSingleImage, selectSingleImageFromGallery } from 'src/helpers';
+import { Heading, MainButton } from '..';
+import styles from './styles';
+
+type AvatarPickerProps = {
+  onPick: (avatar?: Asset) => void;
+};
+
+const AvatarPicker: React.FC<AvatarPickerProps> = ({ onPick }) => {
+  const pick = () => {
```
Comment (yevhenprybytok) [inline src/components/avatar-picker/avatar-picker.tsx]:
```suggestion
  const handlePick = () => {
```

File: src/components/avatar-picker/avatar-picker.tsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    selectSingleImageFromGallery().then(onPick);
+  };
+
+  const capture = () => {
```
Comment (yevhenprybytok) [inline src/components/avatar-picker/avatar-picker.tsx]:
```suggestion
  const handlePress/handleCapture = () => {
```

File: src/helpers/image-capture/multipart-file.ts [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+
+interface IMultipartFile {
+  uri: string;
+  name: string;
+  type: string;
+}
+
+function assetToMultipartFile(asset: Asset): IMultipartFile {
+  return {
+    uri: asset.uri ?? asset.base64 ?? '',
```
Comment (yevhenprybytok) [inline src/helpers/image-capture/multipart-file.ts:11]: shouldn't base64 value be first in here?

**Review by yevhenprybytok — APPROVED** [review]

File: android/settings.gradle (hunk 3 above)
Comment (R-Mielamud) [reply android/settings.gradle:3]: No, it doesn't. I've tested it, everything works fine.

File: src/screens/onboarding/components/step-content/common.ts (hunk 9 above)
Comment (R-Mielamud) [reply src/screens/onboarding/components/step-content/common.ts:7]: Ask @katyakostikova. I think no :)

File: src/helpers/image-capture/multipart-file.ts (hunk 12 above)
Comment (R-Mielamud) [reply src/helpers/image-capture/multipart-file.ts:11]: No

File: src/helpers/image-capture/capture.ts (hunk 8 above)
Comment (R-Mielamud) [reply src/helpers/image-capture/capture.ts]: Fixed

**Pull request merged**